src/components/TodoItem.tsx

```diff
@@ -0,0 +1,214 @@
+    if (updatedTitle.trim() === '') {
+      handleDelete();
+
+      return;
+    }
+
+    if (!isEqualTitle) {
+      setIsUpdating(true);
+
+      updateTodo(todo.id, { title: updatedTitle })
+        .then(() => {
+          const updatedTodos = todos.map((t) => {
+            if (t.id === todo.id) {
+              return { ...t, title: updatedTitle.trim() };
+            }
+
+            return t;
+          });
+
+          setUpdatingTodo(null);
+          setTodos(updatedTodos);
+        })
+        .catch(() => setError(ErrorMessage.UpdateTodo))
+        .finally(() => setIsUpdating(false));
+    } else {
+      setUpdatingTodo(null);
+    }
+  };
+
+  const handleKeyUp = (
+    event: React.KeyboardEvent<HTMLInputElement>,
+  ) => {
+    if (event.key === Key.Enter) {
+      handleBlur();
+    } else if (event.key === Key.Escape) {
+      setUpdatedTitle(todo.title);
+      setUpdatingTodo(null);
+      setIsUpdating(false);
+    }
+  };
+
+  return (
+    <div
+      data-cy="Todo"
+      className={cn('todo', { completed: todo.completed })}
+    >
+      <label
+        className="todo__status-label"
+      >
+        <input
+          data-cy="TodoStatus"
+          type="checkbox"
+          className="todo__status"
+          checked={todo.completed}
+          onChange={() => handleTodoToggle()}
+        />
+      </label>
+      {(updatingTodo !== todo.id)
+        ? (
+          <>
+            <span
+              data-cy="TodoTitle"
+              className="todo__title"
+              onDoubleClick={() => handleDoubleClick()}
+            >
+              {todo.title.trim()}
+            </span>
+
+            <button
+              type="button"
+              className="todo__remove"
+              data-cy="TodoDelete"
+              onClick={() => handleDelete()}
+            >
+              ×
+            </button>
+          </>
+        )
+        : (
+          <form onSubmit={(e) => e.preventDefault()}>
+            <input
+              data-cy="TodoTitleField"
+              type="text"
+              className="todo__title-field"
+              placeholder="Empty todo will be deleted"
+              value={updatedTitle}
+              ref={todoInputRef}
+              onKeyUp={(event) => handleKeyUp(event)}
+              onChange={(event) => setUpdatedTitle(event.target.value)}
+              onBlur={() => handleBlur()}
+            />
+          </form>
+        )}
+
+      <div
+        data-cy="TodoLoader"
+        className={cn(
+          'modal overlay',
+          {
+            'is-active': todo.id === tempTodo?.id
+            || todo.id === deletingTodo?.id
+            || (isClearCompleted && todo.completed)
+            || (isUpdating && updatingTodo === todo.id)
+            || (isToggled && toggledTodos === todo.id)
+            || (isEveryTodoCompleted && isToggleAllClicked && todo.completed)
+            || (!isEveryTodoCompleted && isToggleAllClicked && !todo.completed),
```

Comment:
Here is not a good idea to write the logic in the jsx. Would be better to do it separately